flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java

```diff
@@ -449,6 +486,10 @@ static ErrorResponse readFrom(ByteBuf buffer) throws Exception {
 				}
 			}
 		}
+
+		static int getMessageHeaderLength(int lengthWithoutFrameHeader) {
+			return lengthWithoutFrameHeader;
```

Comment:
Actually this way is still hacky, the specific message should be aware of its length, but it is still understood by the upper layer and take a round to get it.